plugins/unix-socket.js

```diff
@@ -15,8 +15,14 @@ module.exports = function (opts) {
   return {
     name: 'unix',
     scope: function() { return scope },
-    server: !opts.server ? null : function (onConnection, cb) {
-      if(started) return
+    server: function (onConnection, cb) {
+      if (started) return
+
+      if (opts.scope !== "device" && !opts.server) {
+        debug('Insecure scope for unix socket! If you really want this, you need to provide pass a \'server: true\' option')
```

Comment:
this prevents the server being created, but it reintroduces the vulnerability (leaking user name) because below you've removed the check in stringify, to return if server is not set (!)

I think a better solution is to refuse to run if the scope is set to something other than "device" (like either throw, or just revert it to "device")

Reply:
Dang it, you correct. Pushed fix to force device scope.